spec/ruby/core/module/refine_spec.rb

```diff
@@ -762,6 +787,120 @@ def foo
 
       result.should == "foo"
     end
+
+    it "looks only in the refined class even if there is another active refinement" do
+      refined_class = ModuleSpecs.build_refined_class
+
+      refinement = Module.new do
+        refine refined_class do
+          def bar
+            "foo from refinement"
+          end
+        end
+      end
+
+      refinement_with_super = Module.new do
+        refine refined_class do
+          def bar
+            super
+          end
+        end
+      end
+
+      -> {
+        Module.new do
+          using refinement
+          using refinement_with_super
+          refined_class.new.bar
+        end
+      }.should raise_error(NoMethodError)
+    end
+
+    it "looks in the another active refinement if super called from included modules
+        (https://bugs.ruby-lang.org/issues/16977)" do
+      refined_class = Class.new(ModuleSpecs.build_refined_class) do
+        def foo
+          "foo from C"
+        end
+      end
+
+      a = Module.new do
+        def foo
+          "foo from A"
+        end
+      end
+
+      b = Module.new do
+        def foo
+          super
+        end
+      end
+
+      refinement = Module.new do
+        refine refined_class do
+          include a
+        end
+      end
+
+      refinement1 = Module.new do
+        refine refined_class do
+          include b
+        end
+      end
+
+      result = nil
+      Module.new do
+        using refinement
+        using refinement1
+        result = refined_class.new.foo
+      end
+
+      result.should == "foo from A"
+    end
+
+    it "looks in the current active refinement from included modules" do
+      refined_class = Class.new(ModuleSpecs.build_refined_class) do
+        def foo
+          "foo from C"
+        end
+      end
+
+      a = Module.new do
+        def foo
+          "foo from A"
+        end
+      end
+
+      b = Module.new do
+        def foo
+          super
+        end
+      end
+
+      refinement = Module.new do
+        refine refined_class do
+          def foo
+            "I'm activated, but have less priority"
+          end
+        end
+      end
+
+      refinement1 = Module.new do
+        refine refined_class do
+          include a
+          include b
+        end
+      end
+
+      result = nil
+      Module.new do
+        using refinement
+        using refinement1
+        result = refined_class.new.foo
+      end
+
+      result.should == "foo from A"
+    end
```

Comment:
I think these examples would be a lot easier to read if we returned an Array of the module names of the methods called, like `["b", "a", "C"]`. That would also guarantee the implementation anything in between that just does `super`.
Could you use this style in the new specs?